tests/unit/socket_test.cc

```diff
@@ -224,3 +224,29 @@ SEASTAR_TEST_CASE(socket_on_close_local_shutdown_test) {
         when_all(std::move(client), std::move(server)).discard_result().get();
     });
 }
+
+SEASTAR_TEST_CASE(socket_connect_abort_test) {
```

Comment:
Please add a comment explaining what this test does. the comment needs to at least mention the word shutdown(), and I recommend also mentioning #2302 and that it fails with this-and-that backend. A future user who encounters a failure might find this information useful.

Reply:
Done